app-frontend/src/app/components/histogram/nodeHistogram/nodeHistogram.controller.js

```diff
@@ -0,0 +1,236 @@
+/* globals d3 _ */
+/* eslint-disable*/
+const uuid = function b(a){return a?(a^Math.random()*16>>a/4).toString(16):([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g,b)};
+/* eslint-enable */
+
+const defaultData = {minimum: 0, maximum: 255, buckets: _.range(0, 256).map((pixelValue) => {
+    let x = parseInt(pixelValue, 10);
+    let y = x === 0 || x === 255 ? 0 : 1;
+    return [x, y];
+})};
+
+export default class NodeHistogramController {
+    constructor($log, $scope, $element) {
+        'ngInject';
+        this.$log = $log;
+        this.$scope = $scope;
+        this.$element = $element;
+    }
+
+    $onInit() {
+        this.setDefaults(this.options);
+        this.$scope.$watch('$ctrl.minChecked', (minChecked) => {
```

Comment:
I wonder if these two watches could be cleaned up a bit by adding `ng-change` to the `ng-model` directive; I think that would allow you to tie the control directly to `this._masks.min` and skip the first two lines. The remaining lines are the same across both functions so they could be pulled out into a single function that would be called by `ng-change`.

Reply:
yea, this is a little bit messy